File: go/raindrops/raindrops.go
package raindrops

import (
  "strconv"
)

func Convert(n int) string {
  primes := map[int]string{
    3: "Pling",
    5: "Plang",
    7: "Plong",
  }

  result := createResult(n, primes)

  if result == "" {
    return strconv.Itoa(n)
  } else {
    return result
  }
}

func createResult(n int, primes map[int]string) (result string) {
  for prime, saying := range primes {
    if n%prime == 0 {
      result += saying
    }
  }
  return result
}

File: ruby/raindrops/raindrops.rb
class Raindrops

  def convert(number)
    result = droplet_noises.select { |_, verifier| verifier.call(number) }.keys.join
    result = number.to_s if result.empty?
    result
  end

  private

  def droplet_noises
    {
      "Pling" => ->(number) { number.divides_evenly_by?(3) },
      "Plang" => ->(number) { number.divides_evenly_by?(5) },
      "Plong" => ->(number) { number.divides_evenly_by?(7) }
    }
  end

end

class Fixnum

  def divides_evenly_by?(divisor)
    self % divisor == 0
  end

end
